WCOSS2 resource updates

KateFriedman-NOAA opened the pull request:
**Description**

This PR brings in some resource optimization changes for the WCOSS2 port:

1. Update to C768 block in `parm/config/config.fv3.nco.static` to update resource values for gdasfcst. Values come from optimization work between NCO/EMC.
2. Numerous updates to `parm/config/config.resources.nco.static` to fold in resource optimizations covered in ecf script updates and PR #603.
3. Update emc mode resource configs (`config.fv3.emc.dyn` and `config.resources.emc.dyn`) with values from nco mode resource configs. Only updated C384 and C768 settings in `config.fv3.emc.dyn`. This update will support dev/emc mode users running the package on WCOSS2. Can be used as starting point to test package on other platforms for the dev side of the WCOSS2 port.

Refs: #398, #399

**Type of change**

Port updates.

**How Has This Been Tested?**

By rerunning affected jobs in test on Dogwood.

KateFriedman-NOAA committed:
Update analysis job ecf resource settings

- based on optimization testing the gdas and gfs analysis jobs will now
use adjusted resource values in operations
- update ecf PBS resource directive with new optimized settings

Refs: #399

KateFriedman-NOAA committed:
Update for C768 gdasfcst job resource settings

- based on optimization testing the gdasfcst job will use new resource
settings at C768
- the updated resource settings run the gdasfcst with similar cores and
timing compared to WCOSS1 ops

Refs: #399

KateFriedman-NOAA committed:
Numerous resource updates based on optimization

- folding in updates to resources based on NCO/EMC optimization testing
- add some new gfs-specific variables so some jobs run differently in
the gfs suite compared to the gdas suite
- update to ppn value for the gdasfcst job (companion of config.fv3
updates)

Refs: #399

KateFriedman-NOAA committed:
Update resources for gdasesfc job in ecf script

- update the resources in the ecf script for the enkfgdas sfc job based
on optimization testing

Refs: #399

KateFriedman-NOAA committed:
Add COMIN_OBS/COMIN_GES_OBS and related xml support

- add COMIN_OBS/COMIN_GES_OBS setting to config.base.emc.dyn for running
analysis/enkf jobs in emc/dev mode; NCO mode uses compath.py in ops
- COMIN_OBS/COMIN_GES_OBS needs to be set before use when in emc/dev
mode
- to support COMIN_GES_OBS setting in config.base values of GDATE, gPDY,
gcyc, and GDUMP need to be set before config.base is read, therefore have added
GDATE/gPDY/gcyc/GDUMP to generated rocoto xml envvars
- CDUMP/PDY/cyc/CDUMP already in rocoto xml for similar job support
- these changes should not impact NCO mode in ops, only emc/dev mode

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

KateFriedman-NOAA committed:
Cleanup of config.fv3.nco.static

- remove THEIA from machine if-block
- remove extra whitespaces in a couple npe_wav* lines

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'feature/ops-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow into feature/ops-wcoss2

* 'feature/ops-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow:
  Add COMIN_OBS/COMIN_GES_OBS and related xml support

KateFriedman-NOAA committed:
Update config.resource.emc.dyn with tested values

- duplicate updates in config.resources.nco.static into the emc mode
version (config.resources.emc.dyn)
- use tested resources on WCOSS2 as starting point for emc mode

Refs: #399

KateFriedman-NOAA committed:
Update C384 and C768 values in config.fv3.emc.dyn

- fold in tested updates for C384 and C768 on WCOSS2 into emc mode
version (config.fv3.emc.dyn)
- use tested values on WCOSS2 as starting point for dev side of this
port and support dev users in emc mode on WCOSS2 when running ops GFS

Refs: #399

KateFriedman-NOAA committed:
Update analysis job walltime to 50 mins

Refs: #399

WalterKolczynski-NOAA commented on parm/config/config.resources.nco.static:
```diff
@@ -49,89 +49,99 @@ if [ $step = "prep" -o $step = "prepbufr" ]; then
-    export nth_anal=12
-    export npe_anal_gfs=1000
-    export nth_anal_gfs=12
+    export wtime_anal="00:50:00"
```
NCO has 1:00 for gdas and 0:40 for gfs

aerorahul replied:
@KateFriedman-NOAA and I spoke.  We will make them both 50 mins in the PR #603

KateFriedman-NOAA committed:
Optimization resource updates from NCO

- fold in further resource optimization values from testing done by
NCO/Wei and committed to ecf scripts in PR #603

Refs: #399

KateFriedman-NOAA committed:
Merge remote-tracking branch 'upstream/feature/ops-wcoss2' into feature/ops-wcoss2

* upstream/feature/ops-wcoss2:
  Update analysis job walltime to 50mins
  bringing in changes from @WeiWei-NCO after his testing
  Update gdasechgres job resources
  Update esfc and analysis job resources
  update resources for more jobs in ecf scripts from NCO
  update resources for more jobs in ecf scripts from NCO
  update resources for jgfs_atmos_tropcy_qc_reloc.ecf.  Remove developer overwrite section
  update resources for jgdas_atmos_tropcy_qc_reloc.ecf.  Remove developer overwrite section
  update resources for more jobs to include memory in ecf scripts.  wave init jobs need modules for Intel loaded
  update resources for wave jobs to include memory in ecf scripts
  update resources for atmos chgres for enkf in ecf scripts
  update resources for atmos pp wafs_gcip in ecf scripts
  update resources for atmos gempak_meta in ecf scripts
  update resources for atmos gempak in ecf scripts
  update resources for wave init, post and prep jobs in ecf scripts
  fix resource allocations for some jobs that NCO flagged were allocating too many cores

KateFriedman-NOAA committed:
A few resource updates from NCO and WCOSS_C removal

- remove unneeded WCOSS_C memory settings
- remove unneeded THEIA memory reference
- update awips job resources for atmosphere and wave

Refs: #399

aerorahul reviewed (approved):
looks good assuming this was tested.

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2
